src/fragments/DatePickerRange.react.js

```diff
@@ -0,0 +1,194 @@
+import 'react-dates/initialize';
+import {DateRangePicker} from 'react-dates';
+import React, {Component} from 'react';
+import uniqid from 'uniqid';
+
+import {propTypes, defaultProps} from '../components/DatePickerRange.react';
+import convertToMoment from '../utils/convertToMoment';
+
+export default class DatePickerRange extends Component {
+    constructor(props) {
+        super(props);
+        this.propsToState = this.propsToState.bind(this);
+        this.onDatesChange = this.onDatesChange.bind(this);
+        this.isOutsideRange = this.isOutsideRange.bind(this);
+        this.state = {
+            focused: false,
+            start_date_id: props.start_date_id || uniqid(),
+            end_date_id: props.end_date_id || uniqid(),
+        };
+    }
+
+    propsToState(newProps, force = false) {
+        const state = {};
+
+        if (force || newProps.start_date !== this.props.start_date) {
+            state.start_date = newProps.start_date;
+        }
+
+        if (force || newProps.end_date !== this.props.end_date) {
+            state.end_date = newProps.end_date;
+        }
+
+        if (Object.keys(state).length) {
+            this.setState(state);
+        }
+    }
+
+    componentWillReceiveProps(newProps) {
+        this.propsToState(newProps);
+    }
+
+    componentWillMount() {
+        this.propsToState(this.props, true);
```

Comment:
This is the most significant code change. Since the real/fragment component is wrapped, its props lifecycle is slightly different, `propsToState` is updated to only update the state when the value changes or if forced (on mount)

Reply:
I'm missing something - this seems like it would have been a good idea before, but somehow more than a good idea it's actually *necessary* now? What's the bad thing that happens if we don't do this?

Reply:
props override state, always - the component essentially doesn't work at all anymore